indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -2733,6 +2733,10 @@ private boolean updatePartitionDataFromStream()
 
     log.debug("Found [%d] partitions for stream [%s]", partitionIdsFromSupplier.size(), ioConfig.getStream());
 
+    Integer taskCount = spec.getIoConfig().getTaskCount();
```

Comment:
It seems that `spec.getIoConfig().getTaskCount()` will always be non-null.
Maybe we could update the `SeekableStreamSupervisorIOConfig.getTaskCount()` to return a primitive `int` instead of a boxed `Integer` to avoid confusion.

```suggestion
    final int configuredTaskCount = spec.getIoConfig().getTaskCount();
```